Add ReadMe (#42)
wso2/open-healthcare-prebuilt-services: conformance/metadata-fhirr4-service/README.md

# FHIR R4 Metadata Service

## Introduction
This service provides implementation of FHIR Metadata API. This implements 
[capabilities](https://www.hl7.org/fhir/http.html#capabilities) interaction, which is used to retrieve capability 
statement describing the server's current operational functionality by FHIR client applications. 

This FHIR server interaction returns Capability Statement ([CapabilityStatement](http://hl7.org/fhir/StructureDefinition/CapabilityStatement) 
FHIR resource) that specifies which resource types and interactions are supported by the FHIR server

```Supported FHIR version is 4.0.1.```

> ### Setup and run

Reviewer: ```suggestion
## Setup and run
```

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/b1b355e6d8a01038dca36588c1af3cd618f87fa1

1. Clone this repository to your local machine and navigate to the pre-built service on metadata-fhirr4-service.

2. Configure the Config.toml with relevant configurations mentioned in [Configurations](#configurations).

    A sample `Config.toml` is as below.

    ```
    ## server related configurables
    [configFHIRServer]
    version = "1.2.0"
    name = "WSO2OpenHealthcareFHIR"
    title = "FHIR Server"
    status = "active"
    experimental = true
    date = "2022-11-24"
    kind = "instance"
    fhirVersion = "4.0.1"
    format = ["json"]
    patchFormat = ["application/json-patch+json"]
    implementationUrl = "<FHIR_BASE_URL>"
    implementationDescription = "WSO2 Open Healthcare FHIR"

    ## server security related configurables
    [configRest]
    mode = "server"
    resourceFilePath = "resources/fhir_resources.json"
    interaction = ["search-system"]
    [configRest.security]
    cors = false
    discoveryEndpoint = "https://api.asgardeo.io/t/<organization_name>/oauth2/token/.well-known/openid-configuration"
    managementEndpoint = "https://api.asgardeo.io/t/<organization_name>/oauth2/manage"
    ```

3. Run the project.

    ```ballerina
    bal run
    ```

> 4. Invoke the APIs.

Reviewer 2: ```suggestion
4. Invoke the API.
```

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/3c45d53b694704102f3950d7f4431d600fc9e72e

    Sample request for FHIR Capability Statement:

    ```

>     curl --location 'localhost:9090/fhir/r4/metadata'

Reviewer 2: Why do we need --location here? Can we make it simple?

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/3c45d53b694704102f3950d7f4431d600fc9e72e

2. Enter a unique organization name. For example, Stark Industries.
3. Read and accept the privacy policy and terms of use.
4. Click Create.
This creates the organization and opens the Project Home page of the default project created for you.

### Steps to Deploy Metadata Prebuilt Service in Choreo
1. Create Service Component
    * Fork the pre-built Ballerina services repository (https://github.com/wso2/open-healthcare-prebuilt-services) to your Github organization.
    * Follow the official documentation to create and configure a service: https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-ballerina-rest-api/#step-1-create-a-service-component.

    * Once the component creation is complete, you will see the component overview page.

2. Configure and Deploy

    Follow the official documentation to deploy the Metadata prebuilt service to your organization https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-ballerina-rest-api/#step-2-build-and-deploy.

    On deployment configurables mentioned in [Configurations](#configurations) needs to be configured in Choreo configurable editor.

> ### Configurations

Reviewer: ```suggestion
## Configurations
```

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/b1b355e6d8a01038dca36588c1af3cd618f87fa1

# FHIR R4 Metadata Service

## Introduction
This service provides implementation of FHIR Metadata API. This implements 
[capabilities](https://www.hl7.org/fhir/http.html#capabilities) interaction, which is used to retrieve capability 
statement describing the server's current operational functionality by FHIR client applications. 

This FHIR server interaction returns Capability Statement ([CapabilityStatement](http://hl7.org/fhir/StructureDefinition/CapabilityStatement) 
FHIR resource) that specifies which resource types and interactions are supported by the FHIR server

```Supported FHIR version is 4.0.1.```

> 

Reviewer: We missed to add how to build and run the service as a bal user in standlone mode?

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/8b982af37b97ff9cd7f6f9e8a273a8c6a9400010

If you are signing in to the Choreo Console for the first time, create an organization as follows:

1. Go to https://console.choreo.dev/, and sign in using your preferred method.
2. Enter a unique organization name. For example, Stark Industries.
3. Read and accept the privacy policy and terms of use.
4. Click Create.
This creates the organization and opens the Project Home page of the default project created for you.

### Steps to Deploy Metadata Prebuilt Service in Choreo
1. Create Service Component
    * Fork the pre-built Ballerina services repository (https://github.com/wso2/open-healthcare-prebuilt-services) to your Github organization.
    * Follow the official documentation to create and configure a service: https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-ballerina-rest-api/#step-1-create-a-service-component.

    * Once the component creation is complete, you will see the component overview page.

2. Configure and Deploy

    Follow the official documentation to deploy the Metadata prebuilt service to your organization https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-ballerina-rest-api/#step-2-build-and-deploy.

>     Following configurations needs to be configured in a `Config.toml` or in Choreo configurable editor.

Reviewer 2: ```suggestion
    Following configurations need to be added in a `Config.toml` or in the Choreo configurations editor.
```

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/8b982af37b97ff9cd7f6f9e8a273a8c6a9400010

>     | Configuration                | Description                                                                                        |

Reviewer: Shall we have this configurations in a topic so that it can be linked in other places where running locally or deploying in choreo options?

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/8b982af37b97ff9cd7f6f9e8a273a8c6a9400010

    |------------------------------|----------------------------------------------------------------------------------------------------|
    | `version`                    | Business version of the capability statement <br/><br/>  eg: `0.1.7`                               |

>     | `name`                       | Name for this capability statement (computer friendly)  <br/><br/> eg: `WSO2 Open Healthcare FHIR` | 

Reviewer 2: Example cannot have spaces - https://hl7.org/fhir/R4/capabilitystatement-definitions.html#CapabilityStatement.name

Author: Updated in https://github.com/wso2/open-healthcare-prebuilt-services/commit/8b982af37b97ff9cd7f6f9e8a273a8c6a9400010

### Prerequisites

If you are signing in to the Choreo Console for the first time, create an organization as follows:

1. Go to https://console.choreo.dev/, and sign in using your preferred method.
2. Enter a unique organization name. For example, Stark Industries.
3. Read and accept the privacy policy and terms of use.
4. Click Create.
This creates the organization and opens the Project Home page of the default project created for you.

### Steps to Deploy Metadata Prebuilt Service in Choreo
1. Create Service Component
    * Fork the pre-built Ballerina services repository (https://github.com/wso2/open-healthcare-prebuilt-services) to your Github organization.
    * Follow the official documentation to create and configure a service: https://wso2.com/choreo/docs/develop-components/develop-services/develop-a-ballerina-rest-api/#step-1-create-a-service-component.

    * Once the component creation is complete, you will see the component overview page.

> 2. Build and Deploy

Reviewer 2: Can we rename this step to Configure and Deploy? And list the configs here itself.

Author: Fixed in https://github.com/wso2/open-healthcare-prebuilt-services/commit/9d390d3b1e18552398324f86e3bb953423fcfc72
